borg/helpers.py

```diff
@@ -699,6 +705,10 @@ class Buffer:
     """
     provide a thread-local buffer
     """
+
+    class OutOfMemory(Error, OSError):
```

Comment:
my line of thinking was that `Buffer` has a fixed "pool size" so to speak; trying to allocate more memory than that means that there is no more `Buffer` memory available.